### PR TITLE
fix(gateway): add test coverage for GW-0405 AC5 and GW-0808 T-0815a-e

### LINKED ISSUE #745: Maintenance audit: sonde-gateway test coverage gaps
## Summary

Maintenance audit (2026-04-16) identified 2 test coverage gaps in sonde-gateway.

### F-025: GW-0405 initial map data edge cases untested (Medium)
**Category**: D12_ACCEPTANCE_CRITERIA_GAP

AC5 (explicit .maps sections have no initial data) and AC6 (section name prefix matching for .rodata.str1.1) are not confirmed in the test suite.

**Action**: Add tests for AC5 and AC6 edge cases.

### F-028: GW-0808 channel persistence tests missing (Medium)
**Category**: D11_UNTESTED_VALIDATION_ENTRY

T-0815a through T-0815e (ESP-NOW channel persistence lifecycle) are specified but not confirmed as named test functions.

**Action**: Verify or add T-0815a-e tests covering: CLI seed on first startup, SetModemChannel atomic update, modem reconnect from DB, BLE pairing uses DB value, CLI ignored if DB persisted.

## Audit metadata
- Audit date: 2026-04-16
- False positives eliminated: F-007, F-008, F-009, F-023, F-024, F-026, F-027, F-029, F-049

### PR DESCRIPTION
**Pull request by Alan-Jowett**
## Summary

Closes #745 — addresses maintenance audit findings F-025 and F-028 by adding 6 missing tests for `sonde-gateway`.

### F-025: GW-0405 initial map data edge cases (AC5, AC6)

- **AC5**: Added `extract_global_section_data_maps_no_initial_data` unit test confirming that explicit `.maps` sections are excluded from initial data extraction. The test builds an ELF with both `.maps` and `.rodata` sections and asserts only `.rodata` produces initial data.
- **AC6**: Updated doc comment on existing `extract_global_section_data_rodata_prefix` test to explicitly reference AC6.

### F-028: GW-0808 channel persistence lifecycle (T-0815a–e)

Extracted `resolve_espnow_channel()` from the gateway binary into `engine.rs` as a testable library function. The binary now calls this function instead of inlining the seed/load logic.

Added 5 integration tests in `tests/channel_persistence.rs`:

| Test | Validates |
|------|-----------|
| `t0815a_channel_persisted_after_set_modem_channel` | `SetModemChannel(7)` persists `espnow_channel = "7"` in DB |
| `t0815b_modem_reconnect_restores_persisted_channel` | Reconnect resolves channel 7 from DB (not CLI default 1) |
| `t0815c_ble_pairing_uses_persisted_channel` | BLE `REGISTER_PHONE` response carries `rf_channel = 7` from DB |
| `t0815d_cli_channel_seeds_database_on_first_startup` | Fresh DB + `--channel 3` seeds DB with `"3"` |
| `t0815e_persisted_channel_overrides_cli_channel` | Pre-populated DB `"7"` overrides CLI `--channel 3` |

## Testing

- All 6 new tests pass
- Full `sonde-gateway` test suite passes (397+ tests, 0 failures)
- `cargo clippy -p sonde-gateway --tests -- -D warnings` clean

## Files changed

| File | Change |
|------|--------|
| `crates/sonde-gateway/src/engine.rs` | Extract `resolve_espnow_channel()` |
| `crates/sonde-gateway/src/lib.rs` | Re-export `resolve_espnow_channel` |
| `crates/sonde-gateway/src/bin/gateway.rs` | Use extracted function |
| `crates/sonde-gateway/src/program.rs` | Add AC5 test + AC6 doc comment |
| `crates/sonde-gateway/tests/channel_persistence.rs` | New: T-0815a–e tests |
| `crates/sonde-gateway/tests/common/mod.rs` | Add `#[allow(dead_code)]` to shared helper |

### CODE REVIEW

File: crates/sonde-gateway/src/engine.rs [hunk 1]
```diff
@@ -167,6 +167,49 @@ pub enum PendingCommand {
...
+/// # Errors
+///
+/// Returns an error if storage I/O fails, if the CLI-supplied default is
+/// outside the valid WiFi channel range `1..=14`, if the persisted value is
+/// not a valid `u8`, or if the persisted value is outside `1..=14`.
+pub async fn resolve_espnow_channel(storage: &dyn Storage, cli_channel: u8) -> Result<u8, String> {
+    if !(1..=14).contains(&cli_channel) {
+        return Err(format!(
+            "invalid CLI espnow_channel `{cli_channel}`: expected 1..=14"
+        ));
```
[the remark this answers is not in the document]
Comment (Alan-Jowett) [reply crates/sonde-gateway/src/engine.rs:184]: This function has a single call site (gateway binary startup) where the error is immediately converted to a display string. Adding a typed error enum for a small helper with 3 error paths would be over-engineering. The crate uses typed errors for domain APIs (`StorageError`, `TransportError`) but simple `String` for configuration/startup helpers — consistent with existing patterns.

File: crates/sonde-gateway/src/engine.rs [hunk 2]
```diff
@@ -167,6 +167,49 @@ pub enum PendingCommand {
...
+///
+/// Returns the channel to use for the modem startup handshake.
+///
+/// # Errors
+///
+/// Returns an error if storage I/O fails, if the CLI-supplied default is
+/// outside the valid WiFi channel range `1..=14`, if the persisted value is
+/// not a valid `u8`, or if the persisted value is outside `1..=14`.
+pub async fn resolve_espnow_channel(storage: &dyn Storage, cli_channel: u8) -> Result<u8, String> {
+    if !(1..=14).contains(&cli_channel) {
```
[the remark this answers is not in the document]
Comment (Alan-Jowett) [reply crates/sonde-gateway/src/engine.rs:181]: This is a pre-existing inconsistency not introduced by this PR. The `1..=13` range in `parse_pairing_request` validates incoming node pairing data (nodes use channels 1-13 per the node requirements). The `1..=14` range in admin RPCs and `resolve_espnow_channel` is the gateway-side config range matching `UsbEspNowTransport::change_channel`. Unifying these ranges is a separate discussion — filing as a follow-up would make sense but it's out of scope for a test-coverage PR.

File: crates/sonde-gateway/tests/channel_persistence.rs [hunk 3]
```diff
@@ -0,0 +1,256 @@
+// SPDX-License-Identifier: MIT
+// Copyright (c) 2026 sonde contributors
+
+//! ESP-NOW channel persistence tests (T-0815a through T-0815e).
+//!
+//! Validates GW-0808: the gateway persists the ESP-NOW channel in the database
+//! so that the value survives gateway and modem restarts.
```
[the remark this answers is not in the document]
Comment (Alan-Jowett) [reply crates/sonde-gateway/tests/channel_persistence.rs:7]: SQLite persistence round-trips are already covered by `sqlite_storage.rs` unit tests (`test_config_round_trip`). These tests validate the T-0815a-e procedures specified in the validation plan, which describe component behavior against the `Storage` trait — `InMemoryStorage` is the correct abstraction level. Adding SQLite-specific tests would be scope creep beyond issue #745.

File: crates/sonde-gateway/tests/channel_persistence.rs [hunk 4]
```diff
@@ -0,0 +1,256 @@
...
+use sonde_protocol::{encode_ble_envelope, parse_ble_envelope};
+
+use sonde_gateway::admin::pb::gateway_admin_server::GatewayAdmin;
+use sonde_gateway::admin::pb::*;
+use tonic::Request;
+
+use common::{build_admin_with_modem, read_modem_msg};
+
+const BLE_MSG_REGISTER_PHONE: u8 = 0x02;
+const BLE_MSG_PHONE_REGISTERED: u8 = 0x82;
```
[the remark this answers is not in the document]
Comment (Alan-Jowett) [reply crates/sonde-gateway/tests/channel_persistence.rs:28]: This follows the established pattern — the existing `ble_pairing.rs` and `ble_pairing_relay.rs` test files define the same constants locally. Exporting them from the library would require a public API change unrelated to this PR.

File: crates/sonde-gateway/src/program.rs [hunk 5]
```diff
@@ -1309,6 +1309,124 @@ mod tests {
...
+    /// (`.rodata`, `.data`, `.bss`).  An explicit `.maps` section MUST NOT
+    /// appear in the result — map definitions carry structure, not initial
+    /// data.
+    #[test]
+    fn extract_global_section_data_maps_no_initial_data() {
+        // Build an ELF with both `.maps` and `.rodata` sections.
+        // Only `.rodata` should appear in the extracted initial data.
+        let rodata_content = vec![0xDE, 0xAD];
+        let bpf_code: [u8; 16] = [
+            0xb7, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, // mov r0, 0
```
[the remark this answers is not in the document]
Comment (Alan-Jowett) [reply crates/sonde-gateway/src/program.rs:1324]: This test needs a unique ELF with both `.maps` AND `.rodata` sections — neither `make_bpf_elf_with_rodata` (only `.rodata`) nor `make_minimal_bpf_elf_with_maps` (only `.maps`) covers this case. The existing test suite builds custom ELFs inline per test case; this follows that pattern. Generalizing the builder for a single test would add complexity without benefit.
